pkg/kubelet/apis/config/validation/validation.go

```diff
@@ -126,6 +127,16 @@ func ValidateKubeletConfiguration(kc *kubeletconfig.KubeletConfiguration) error
 	if kc.TopologyManagerPolicy != kubeletconfig.NoneTopologyManagerPolicy && !localFeatureGate.Enabled(features.TopologyManager) {
 		allErrors = append(allErrors, fmt.Errorf("invalid configuration: topologyManagerPolicy %v requires feature gate TopologyManager", kc.TopologyManagerPolicy))
 	}
+	if kc.RegisterNode && len(kc.RegisterWithTaints) > 0 {
```

Comment:
drop `if kc.RegisterNode && len(kc.RegisterWithTaints) > 0 {`

we want to check any provided `RegisterWithTaints` values regardless of whether RegisterNode was specified, and range will no-op on a nil or zero-length list